hazelcast/src/main/java/com/hazelcast/map/impl/proxy/MapProxySupport.java

```diff
@@ -827,8 +827,9 @@ protected void putAllInternal(final Map<? extends Object, ? extends Object> entr
         int partitionCount = partitionService.getPartitionCount();
         boolean tooManyEntries = entries.size() > (partitionCount * factor);
         try {
+            List<Future<?>> futures = new LinkedList<Future<?>>();
+
```

Comment:
I'm not following the flow of the MapProxySupport.putAllInternal method. 

If there are many operations, the system will merge the operations per partition. But if there are not enough operations, the system will not do so. 

AFAIK it is good to do always because making remote calls is 'extremely' expensive compared to the amount of work to be done. Imho we should get rid of the complete (!tooManyEntries) branch and always execute initial branch.

So perhaps your code isn't needed at all.